airbyte-cdk/python/airbyte_cdk/entrypoint.py

```diff
@@ -82,7 +82,8 @@ def run(self, parsed_args: argparse.Namespace) -> Iterable[str]:
         with tempfile.TemporaryDirectory() as temp_dir:
             if cmd == "spec":
                 message = AirbyteMessage(type=Type.SPEC, spec=source_spec)
-                yield message.json(exclude_unset=True)
+                yield from [self.airbyte_message_to_string(queued_message) for queued_message in self._emit_queued_messages(self.source)]
+                yield self.airbyte_message_to_string(message)
             else:
                 raw_config = self.source.read_config(parsed_args.config)
                 config = self.source.configure(raw_config, temp_dir)
```

Comment:
If we really want to preserve order, we could call `_emit_queued_messages`. For example, do we assume that `source.read_config` and `source.configure` can emit messages?

Reply:
Just to clarify, this seems only an issue if other part of the code emits message as print